addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3187,7 +3187,11 @@ var SnippetsMenu = Widget.extend({
             $(range.commonAncestorContainer).parent('[data-oe-model]:not([data-oe-type="html"]):not([data-oe-field="arch"])').length ||
             (e && $(e.target).closest('.fa, img').length ||
             this.options.wysiwyg.lastMediaClicked && $(this.options.wysiwyg.lastMediaClicked).is('.fa, img')) ||
-            (this.options.wysiwyg.lastElement && !this.options.wysiwyg.lastElement.isContentEditable)
+            // An additional test to check if the last element is a link on top
+            // of being editable is mandatory as the editor temporarily makes
+            // them not editable on mouse down.
+            (this.options.wysiwyg.lastElement && this.options.wysiwyg.lastElement.tagName !== 'A'
+                && !this.options.wysiwyg.lastElement.isContentEditable)
```

Comment:
The whole original condition seems ultra messy :grimacing: but well... Anyways, not sure to understand this change as I don't understand the flow you describe in the commit message, can you re-explain or show me ? ;) This is probably better than the other fix you made though, however the other fix fixed something else: 
![Screenshot from 2021-09-20 18-08-32](https://user-images.githubusercontent.com/10338094/134035985-2631f1bb-3783-46ff-b531-966a6402a914.png)
-> Trying to edit a really non-editable link -> the popover still is shown (ok) -> but the edit button is shown and clicking on it crashes (in this case, the fix should be to hide the button or at worst the popover I guess... cc @rdeodoo

Reply:
That's some links we never detected and that should be flagged as non editable. I am afraid there is no discriminant for this (kind of) link(s), so we should just add `o_no_link_popover` on those links from the template.

Reply:
After investigating a bit more, the bug happens when clicking on a link that is displayed behind the edit_menu popover. 
Here is a video

https://user-images.githubusercontent.com/58561718/134140565-69d9d3ae-a239-4883-a256-f3d21f60c878.mp4

In the end I am not sure it is directly linked to the editor setting the link contenteditable attribute to false then to true to correctly display the cursor... 
Reverting the commit 4eff35ab31bfc45b4c8dca8392e9da4f4e2a7afd removes the bug, so it may be related to using "focus" as a trigger for the popover... I am still trying to understand what bootstrap is doing in that case

Reply:
The changes about the popover (manual trigger) LGTM